src/website/app/pages/PaletteDemo/Picker.js

```diff
@@ -0,0 +1,275 @@
+      width: `calc(50% - ${theme.space_inline_sm})`,
+      '&:hover': {
+        transform: 'scale(1.05) rotateX(0deg) translate3d(0, 0, 0)'
+      },
+      '&:nth-child(even)': {
+        marginRight: 0
+      }
+    };
+
+    if (active) {
+      css['&:after'] = {
+        position: 'absolute',
+        top: -18,
+        right: -18,
+        content: `""`,
+        width: 30,
+        height: 30,
+        backgroundColor: 'white',
+        transform: 'rotate(45deg)'
+      };
+    }
+
+    return css;
+  },
+  optionList: ({ collapsed }) => ({
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    display: collapsed ? 'none' : 'block'
+  }),
+  root: {
+    perspective: '800px'
+  },
+
+  swatch: ({ theme, activeColor, collapsed }) => {
+    const hue = color[`${activeColor}_60`];
+    return {
+      color: readableColor(hue),
+      backgroundColor: hue,
+      borderRadius: theme.borderRadius_1,
+      cursor: 'pointer',
+      display: collapsed ? 'block' : 'none',
+      fontSize: theme.fontSize_h3,
+      padding: theme.space_inset_md,
+      textTransform: 'capitalize',
+      [theme.bp_mobile]: {
+        fontSize: theme.fontSize_h5
+      }
+    };
+  },
+
+  title: ({ theme }) => ({
+    marginTop: 0,
+    cursor: 'pointer',
+    [theme.bp_mobile]: {
+      margin: `0 0 ${theme.space_stack_sm}`
+    }
+  })
+};
+
+const Root = createStyledComponent('div', styles.root);
+const CollapseIcon = createStyledComponent(
+  IconKeyboardArrowDown,
+  styles.collapseIcon
+);
+const OptionList = createStyledComponent('div', styles.optionList);
+const Option = createStyledComponent(Paragraph, styles.option);
+const Swatch = createStyledComponent('div', styles.swatch);
+const Title = createStyledComponent(Heading, styles.title);
+
+// for growing the height of the picker
+const Grow = ({ in: inProp, children }: GrowProps) => (
+  <Transition
+    in={inProp}
+    timeout={{ enter: duration - 50, exit: duration / 2 }}>
+    {state => (
+      <div
+        style={{
+          ...styles.grow,
+          ...styles.grow[state]
+        }}>
+        {children}
+      </div>
+    )}
+  </Transition>
+);
+
+// entrance transition for the cards in the expanded menu
+const Flip = ({
+  activeColor,
+  colorName,
+  handleColorChange,
+  readable,
+  in: inProp
+}: FadeProps) => (
+  <Transition in={inProp} timeout={duration}>
+    {state => {
+      return (
+        <Option
+          style={{ ...styles.flip, ...styles.flip[state] }}
+          variant="mouse"
+          active={activeColor === colorName}
+          name={colorName}
+          readableColor={readable}
+          onClick={() => handleColorChange(colorName)}>
+          {colorName}
+          <br />
+          {color[`${colorName}_60`]}
+        </Option>
+      );
+    }}
+  </Transition>
+);
+
+export default class Picker extends React.Component {
+  state = {
+    collapsed: true,
+    visibleThemes: 0,
+    activeColor: 'blue'
```

Comment:
Seems like this could come from a prop that is based on `index`'s state?

Reply:
addressed